### PR TITLE
Displays top used skills on home page

### PR DESCRIPTION
**Pull request by akshatnitd**
First steps for issue #800 

Changes: [Add here what changes were made in this issue and if possible provide links.]
* Displays top used skills on home page

Surge Deployment Link: https://pr-854-fossasia-susi-skill-cms.surge.sh

Screenshots for the change:
screenshot from 2018-06-24 22-44-56

### CODE REVIEW

**Comment by akshatnitd** [comment]
Can be merged only after #852

File: src/components/BrowseSkill/BrowseSkill.js [hunk 1]
```diff
@@ -308,10 +310,68 @@ export default class BrowseSkill extends React.Component {
...
+      urls.API_URL +
+      '/cms/getSkillList.json?group=All&applyFilter=true&filter_name=descending&filter_type=usage&count=10';
+    let self = this;
+    $.ajax({
+      url: url,
+      dataType: 'jsonp',
+      jsonp: 'callback',
+      crossDomain: true,
+      success: function(data) {
+        if (self.state.searchQuery.length > 0) {
```
Comment (anshumanv) [inline src/components/BrowseSkill/BrowseSkill.js]: @akshatnitd you can remove the search query filter. :+1:

**Comment by AnupKumarPanwar** [comment]
The deployment link is not matching the screenshot.

File: src/components/BrowseSkill/BrowseSkill.js (hunk 1 above)
Comment (akshatnitd) [reply src/components/BrowseSkill/BrowseSkill.js]: Yes, doing it.

**Comment by akshatnitd** [comment]
@anshumanv Removed the search filter.

@AnupKumarPanwar @Akshat-Jain It happens some time that, one of the section Top Rated/Top Used/Category Wise Skills do not appear as there is a 503 error from server on API call. We need to resolve this issue on the server. Also, I have opened up an issue for error handling on AJAX calls (#857).

**Comment by AnupKumarPanwar** [comment]
The deployment is not working otherwise code looks fine.

**Comment by akshatnitd** [comment]
@AnupKumarPanwar Can you share a screenshot of your home screen as well as the console section?

**Comment by mariobehling** [comment]
Please provide working test deployment.

screenshot from 2018-06-25 22-00-02

**Comment by anshumanv** [comment]
@akshatnitd It was working initially when I checked but now even I'm getting server error 503 while fetching top used skills, please take a look.

image

**Comment by akshatnitd** [comment]
@mariobehling @anshumanv I had checked, 503 is sent from server incase the frequency of API calls made is too high. It is not a client issue. The deployment links works, when the 503 is not thrown by the server. Multiple refreshes might help. We need to look into this issue and solve it on server. Also, I think when we move the category wise cards to the Category Page, this 503 issue should be solved.

**Comment by akshatnitd** [comment]
We can first solve #867, and I can we can overcome this issue.

**Comment by anshumanv** [comment]
@akshatnitd will solve it today.

File: src/components/BrowseSkill/BrowseSkill.js [hunk 2]
```diff
@@ -299,6 +299,31 @@ export default class BrowseSkill extends React.Component {
       },
     });
   };
+
+  loadTopUsed = () => {
+    let url;
+    url =
+      urls.API_URL +
+      '/cms/getSkillList.json?group=All&applyFilter=true&filter_name=descending&filter_type=usage&count=10';
```
Comment (AnupKumarPanwar) [inline src/components/BrowseSkill/BrowseSkill.js]: You also need to send a duration parameter along with this API, like duration=7 for most used skills in this week.

Comment (akshatnitd) [reply src/components/BrowseSkill/BrowseSkill.js]: @AnupKumarPanwar The PR deals with showing the most used skills in general. For a week's data, user can do that from the sidebar.

**Comment by anshumanv** [comment]
@akshatnitd the cards look good! But I think there's some server issue, the first top used skill that I opened has 0 skill usage count, can you cross check, please?

**Comment by akshatnitd** [comment]
@anshumanv Yes, good catch. Just checked the server code, the duration parameter is set as 0 as default. Opening an issue and solving it.

**Pull request merged**